lms/db/__init__.py

```diff
@@ -28,34 +28,6 @@ def columns(cls):
             if isinstance(property, ColumnProperty)
         ]
 
-    def update_from_dict(self, data, skip_keys=None):
-        """
-        Update this model from the provided dict.
-
-        Any keys listed in ``skip_keys`` will *not* be updated, even if they are in ``data``.
-
-        :param data: The data to update
-        :param skip_keys: A set of keys to exclude from being updated (default: {"id"})
-        :type skip_keys: set[str]
-
-        :raise TypeError: if skip_keys isn't a set
-        """
-
-        if skip_keys is None:
-            skip_keys = {"id"}
-
-        if not isinstance(skip_keys, set):
-            raise TypeError(
-                f"Expected a set of keys to skip but found '{type(skip_keys)}'"
-            )
-
-        columns = set(self.columns())
```

Comment:
Can `BaseClass.columns()` also be removed now?

Reply:
If we could get rid of `Columns` then the entire `BaseClass` could go. The only other thing it has is a `__repr__()`, which we don't need. If we really want to automatically get a nice `__repr__()` for each model class (without having to write it separately for each class) then we can use SQLAlchemy 2's dataclasses support which gives you that

Reply:
We still use the columns on the other side of this group_info setup:

https://github.com/hypothesis/lms/blob/grouping-info-service/lms/resources/_js_config/__init__.py#L574

We could also list all the columns there as a first step to remove all of this.